File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.swift
import Foundation

var ans = 0, cnt = 0, ansWord = ""
var words = ["A", "E", "I", "O", "U"]

func sol(_ str: String) {
    var str = str
    
    if str == ansWord { 
        ans = cnt
        return 
    }
    cnt += 1
    
    if str.count == 5 { return }
    
    for i in 0..<5 {
        str += words[i]
        sol(str)
        str.removeLast()
    }
}

func solution(_ word: String) -> Int {
    ansWord = word
    
    sol("")
    
    return ans
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>

using namespace std;

char alphabets[5] = {'A', 'E', 'I', 'O', 'U'};
int cnt = 0, ans = 0;

void find_word(string cur, string word) {
    if(cur == word) {
        ans = cnt;
        return;
    }
    
    cnt++;
    
    if(cur.size() == 5) return;
    
    for(auto alphabet: alphabets) {
        find_word(cur + alphabet, word);
    }
}

int solution(string word) {
    find_word("", word);
    
    return ans;
}